mysite/chat/views.py

```diff
@@ -68,46 +70,61 @@ def get_will_learn_need_know(self, unit, courseUnit):
                         contaner.add((title, url))
             return will_learn, need_to_know
 
+    def check_course_not_published(self, request, courseUnit):
+        return (not courseUnit.is_published() and
+            not User.objects.filter(
+                id=request.user.id,
+                role__role=Role.INSTRUCTOR,
+                role__course=courseUnit.course
+            ).exists())
+
+    def user_enrolled(self, request, courseUnit):
+        return Role.objects.filter(
+            user=request.user.id, course=courseUnit.course, role=Role.ENROLLED
+        )
+
+    def create_new_chat(self, request, enroll_code, courseUnit):
+        chat = Chat(
+            user=request.user,
+            enroll_code=enroll_code,
+            instructor=courseUnit.course.addedBy,
+            is_preview=False
+        )
+        chat.save(request)
+        return chat
+
+    def get_unitlessons_with_order_null(self, unit):
+        return unit.unitlesson_set.filter(order__isnull=False).exists()
+
+    def get_chat(self, request, enroll_code, **kwargs):
+        return Chat.objects.filter(enroll_code=enroll_code, user=request.user, **kwargs).first()
+
     def get(self, request, enroll_key):
         enroll_code = self.get_enroll_code_object(enroll_key)
         courseUnit = enroll_code.courseUnit
         unit = courseUnit.unit
-        if not unit.unitlesson_set.filter(order__isnull=False).exists():
+        if not self.get_unitlessons_with_order_null(unit):
             return render(
                 request,
                 'lti/error.html',
                 {'message': 'There are no Lessons to display for that Courselet.'}
             )
-        if (
-            not courseUnit.is_published() and
-            not User.objects.filter(
-                id=request.user.id,
-                role__role=Role.INSTRUCTOR,
-                role__course=courseUnit.course
-            ).exists()
-        ):
+        if self.check_course_not_published(request, courseUnit):
             return render(
                 request,
                 'lti/error.html',
                 {'message': 'This Courselet is not published yet.'}
             )
-        if not Role.objects.filter(
-            user=request.user.id, course=courseUnit.course, role=Role.ENROLLED
-        ):
+        if not self.user_enrolled(request, courseUnit):
             enrolling = Role.objects.get_or_create(user=request.user,
                                                    course=courseUnit.course,
                                                    role=Role.SELFSTUDY)[0]
```

Comment:
I just refactored this part of code - moved some parts from function to outer functions to make it replaceable.